### PR TITLE
[SPARK-19912][SQL][NOT FOR MERGE] String literals should be escaped for Hive metastore partition pruning

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

This is not for merging.
This shows the failures of escaping string literals in order support #17266 .
Please see #17266 .

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
@gatorsmile . This is the PR for showing failure.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala [hunk 1]
```diff
@@ -566,13 +566,17 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
...
+        s"""${a.name} ${op.symbol} ${getEscapedString(v.toString)}"""
       case op @ BinaryComparison(Literal(v, _: StringType), a: Attribute)
           if !varcharKeys.contains(a.name) =>
-        s""""$v" ${op.symbol} ${a.name}"""
+        s"""${getEscapedString(v.toString)} ${op.symbol} ${a.name}"""
     }.mkString(" and ")
   }
 
+  private def getEscapedString(str: String): String = {
+    s""""${str.replaceAll("\"", "\\\"")}""""
```
Comment (dongjoon-hyun) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala]: If this is different from what you thought, please let me know~

**Comment by dongjoon-hyun** [comment]
This fails as expected.

**Comment by dongjoon-hyun** [comment]
Retest this please

**Comment by gatorsmile** [comment]
I see. This is a bug of Hive. See the JIRA https://issues.apache.org/jira/browse/HIVE-2943

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala [hunk 2]
```diff
@@ -566,13 +566,17 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
...
       case op @ BinaryComparison(Literal(v, _: StringType), a: Attribute)
           if !varcharKeys.contains(a.name) =>
-        s""""$v" ${op.symbol} ${a.name}"""
+        s"""${getEscapedString(v.toString)} ${op.symbol} ${a.name}"""
     }.mkString(" and ")
   }
 
+  private def getEscapedString(str: String): String = {
+    s""""${str.replace("\"", "\\\"")}""""
+  }
```
Comment (gatorsmile) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala]: I see. This is a bug of Hive. See the JIRA https://issues.apache.org/jira/browse/HIVE-2943

Comment (dongjoon-hyun) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala]: Thank you for clarifying that!

**Pull request closed without merging**